File: inventory-management/src/main/java/com/inventory_management/service/ProfileService.java
package com.inventory_management.service;

import com.inventory_management.dto.profile.ProfileResponseDTO;
import com.inventory_management.dto.profile.UpdateProfileRequestDTO;
import com.inventory_management.dto.profile.ChangePasswordRequestDTO;
import org.springframework.web.multipart.MultipartFile;


public interface ProfileService {
    ProfileResponseDTO getMyProfile();

    ProfileResponseDTO updateMyProfile(
            UpdateProfileRequestDTO request
    );

    void changePassword(
            ChangePasswordRequestDTO request
    );

    ProfileResponseDTO uploadProfilePicture(MultipartFile file);

    void deleteProfilePicture();
}

File: inventory-management-frontend/src/services/profileService.js
import axiosInstance from "../api/axiosConfig";

const profileService = {

    async getMyProfile() {

        const response = await axiosInstance.get(
            "/profile/me"
        );

        return response.data;
    },


    async updateMyProfile(data) {

        const response = await axiosInstance.put(
            "/profile/me",
            data
        );

        return response.data;
    },


    async changePassword(data) {

        await axiosInstance.put(
            "/profile/change-password",
            data
        );

    },


    async uploadProfilePicture(file) {

    const formData = new FormData();

    formData.append("file", file);

    const response = await axiosInstance.post(
        "/profile/me/picture",
        formData,
        {
            headers: {
                "Content-Type": "multipart/form-data"
            }
        }
    );

    return response.data;
},

    async deleteProfilePicture() {

        await axiosInstance.delete(
            "/profile/me/picture"
        );

    }

};

export default profileService;